controllers/datadogagent/feature/apm/feature.go

```diff
@@ -125,6 +133,11 @@ func (f *apmFeature) Configure(dda *v2alpha1.DatadogAgent) (reqComp feature.Requ
 				},
 			},
 		}
+
+		if f.shouldEnableLanguageDetection() {
```

Comment:
Should `f.shouldEnableLanguageDetection()` be called after the `if apm.SingleStepInstrumentation != nil` if block below? I think `f.singleStepInstrumentation` will always be `nil` on this line